src/main/java/uk/nhs/hee/tis/trainee/sync/config/RedisConfig.java

```diff
@@ -0,0 +1,109 @@
+ * NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+ * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package uk.nhs.hee.tis.trainee.sync.config;
+
+import io.lettuce.core.RedisClient;
+import io.lettuce.core.RedisURI;
+import java.time.Duration;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.cache.annotation.CachingConfigurerSupport;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.data.redis.cache.RedisCacheConfiguration;
+import org.springframework.data.redis.connection.lettuce.LettuceConnectionFactory;
+import org.springframework.data.redis.core.RedisTemplate;
+import org.springframework.data.redis.serializer.GenericJackson2JsonRedisSerializer;
+import org.springframework.data.redis.serializer.RedisSerializationContext;
+
+@Configuration
+public class RedisConfig extends CachingConfigurerSupport {
+
+  @Value("${spring.redis.host}")
+  String host;
+
+  @Value("${spring.redis.port}")
+  Integer port;
+
+  @Value("${spring.redis.user}")
+  String user;
+
+  @Value("${spring.redis.password}")
+  char[] password;
+
+  @Value("${spring.redis.timeout}")
+  Long timeout;
+
+  /** Note that the equivalent requests-cache configurations are imported into
+   * {@link uk.nhs.hee.tis.trainee.sync.service.RequestCacheService}
+   */
+  @Value("${spring.redis.time-to-live}")
+  Long dataTtl;
+
+  /**
+   * Configuration for the requests cache.
+   *
+   * @return a Lettuce RedisClient
+   */
+  @Bean
+  public RedisClient getRedisClient() {
+    RedisURI redisUri = new RedisURI();
+    redisUri.setHost(host);
+    redisUri.setPort(port);
+    redisUri.setPassword(password);
+    redisUri.setUsername(user);
+    redisUri.setTimeout(Duration.ofSeconds(timeout));
+    return RedisClient.create(redisUri);
+  }
+
+  /**
+   * Configuration for the general data accessor.
+   *
+   * @return a RedisTemplate
+   */
+  @Bean
+  public RedisTemplate<String, String> redisTemplate(
+      LettuceConnectionFactory lettuceConnectionFactory) {
+
+    RedisTemplate<String, String> redisTemplate = new RedisTemplate<>();
+
+    redisTemplate.setKeySerializer(new GenericJackson2JsonRedisSerializer());
+
+    redisTemplate.setConnectionFactory(lettuceConnectionFactory);
+
+    return redisTemplate;
+  }
+
+  /**
+   * Configuration for the data cache.
+   *
+   * @return a RedisCacheConfiguration
+   */
+  @Bean
+  public RedisCacheConfiguration cacheConfiguration() {
+
+    return RedisCacheConfiguration.defaultCacheConfig()
+        .entryTtl(Duration.ofMinutes(dataTtl))
```

Comment:
Its a bit clumsy to have to do this here, instead of it being read directly from the spring.redis.* config (as the database index is), but I couldn't find a way to do this.